tests/include/test/psa_crypto_helpers.h

```diff
@@ -125,17 +126,21 @@ const char *mbedtls_test_helper_is_psa_leaking(void);
 
 /** Shut down the PSA Crypto subsystem, allowing persistent keys to survive.
  * Expect a clean shutdown, with no slots in use.
+ * mbedtls_psa_random_free() is called before any check for remaining open
+ * keys because when AES_C is not defined, CTR_DRBG relies on PSA to perform
+ * AES-ECB so it holds an open AES key for that since psa_crypto_init().
  *
  * If some key slots are still in use, record the test case as failed and
  * jump to the `exit` label.
  */
 #define PSA_SESSION_DONE()                                             \
-    do                                                                  \
-    {                                                                   \
+    do                                                                 \
+    {                                                                  \
+        mbedtls_psa_random_free();                                     \
```

Comment:
Ok, this works, but it's incompatible with client-server separation (since `mbedtls_psa_random_free` is not exposed client-side). We'd like to get closer to having working tests with client-server separation. So I think we should not do this.

See https://github.com/Mbed-TLS/mbedtls/pull/8521/files#r1403455446 for a different approach.